### PR TITLE
Remove unsupported config condition if config is corrected

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: Wei Liu <liuweixa@redhat.com>

### CODE REVIEW

File: pkg/addonmanager/controllers/clustermanagement/controller.go [hunk 1]
```diff
@@ -126,6 +126,13 @@ func (c *clusterManagementController) sync(ctx context.Context, syncCtx factory.
 		return utils.PatchAddonCondition(ctx, c.addonClient, addonCopy, addon)
 	}
 
+	// Remove unsupported configuration condition if configuration becomes corrected
+	unsupportedConfigCondition := meta.FindStatusCondition(addon.Status.Conditions, UnsupportedConfigurationType)
+	if unsupportedConfigCondition != nil {
+		meta.RemoveStatusCondition(&addonCopy.Status.Conditions, UnsupportedConfigurationType)
```
Comment (qiujian16) [inline pkg/addonmanager/controllers/clustermanagement/controller.go]: why we remove this condition instead of setting it to false?

Comment (skeeey) [reply pkg/addonmanager/controllers/clustermanagement/controller.go]: updated, it better to set it to false than removing it

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**